iceoryx_hoofs/include/iceoryx_hoofs/memory/relative_pointer.inl

```diff
@@ -174,9 +161,11 @@ inline bool RelativePointer<T>::unregisterPtr(const segment_id_t id) noexcept
 template <typename T>
 // NOLINTJUSTIFICATION NewType size is comparable to an integer, hence pass by value is preferred
 // NOLINTNEXTLINE(performance-unnecessary-value-param)
-inline typename RelativePointer<T>::ptr_t RelativePointer<T>::getBasePtr(const segment_id_t id) noexcept
+inline T* RelativePointer<T>::getBasePtr(const segment_id_t id) noexcept
 {
-    return static_cast<RelativePointer<T>::ptr_t>(getRepository().getBasePtr(static_cast<segment_id_underlying_t>(id)));
+    // AXIVION Next Construct AutosarC++19_03-M5.2.8 : Cast to the underyling pointer type is safe as this is
+    // encapsulated in the RelativePointer class and type safety is ensured by using templates
+    return static_cast<T*>(getRepository().getBasePtr(static_cast<segment_id_underlying_t>(id)));
```

Comment:
This could also be `ptr_t` instead of `T*` or `RelativePointer<T>::ptr_t`

Reply:
See [my comment above](https://github.com/eclipse-iceoryx/iceoryx/pull/1765#discussion_r1013754839).